lib/processors/jsdoc/lib/ui5/plugin.js

```diff
@@ -516,6 +523,192 @@ function isPotentialEnum(node) {
+function resolveFullyQunatifiedName(node) {
+	// The missing part is on the left side. The right side is clear.
+	// We would eiter ways would resolve to the same leftmost token.
+	let leftMostName = getLeftmostName(node);
+	let name = getObjectName(node) || "";
+	let originalName = name;
+	const currentScope = getEnclosingVariableScope(node);
+
+	if (!currentScope) {
+		return null;
+	}
+
+	// Checks whether a var has been renamed while destructuring i.e. {A: b} = SomeObject
+	const checkVarRenaming = (variable) => {
+		// variable.defs[0].node.id.type === Syntax.ObjectPattern -> Renaming
+		// variable.defs[0].node.id.properties[0].key.name === variable.name; // Original
+		// variable.defs[0].node.id.properties[0].value.name === variable.name; // Renamed
+
+		return variable.defs
+			.filter(
+				(def) =>
+					def.node &&
+					def.node.id &&
+					def.node.id.type === Syntax.ObjectPattern
+			) // Filter on object patterns => Detructure renamig
+			.map((def) => def.node.id) // Drill down to the ObjectPattern Node
+			.filter((objPatternNode) =>
+				objPatternNode.properties.some(
+					(prop) => prop.value.name === variable.name
+				)
+			) // Filter on ObjectPattern nodes that contain the variable
+			.reduce((acc, objPatternNode) => {
+				return acc.concat(
+					objPatternNode.properties.filter(
+						(prop) => prop.value.name === variable.name
+					)
+				);
+			}, [])
+			.map((prop) => {
+				return { original: prop.key.name, renamed: prop.value.name };
+			})[0];
+	};
+
+	// TODO: Check for hierarchical destructuring
+	// i.e. let {Foo} = myObject; let {Bar} = Foo; --> Bar
+	while (leftMostName) {
+		let potentialChunk = currentScope.variables
+			.filter((curVar) => curVar.name === leftMostName)
+			.reduce((acc, curVar) => {
+				const potentialRenaming = checkVarRenaming(curVar);
+				const filteredValues = curVar.references
+					.filter((ref) => ref.writeExpr)
+					.map((ref) => getObjectName(ref.writeExpr));
+
+				acc.push({
+					ref: filteredValues[0],
+					original: potentialRenaming
+						? potentialRenaming.original
+						: curVar.name,
+					renamed: potentialRenaming
+						? potentialRenaming.renamed
+						: null,
+				});
+				return acc;
+			}, [])[0];
+
+		leftMostName = potentialChunk && potentialChunk.ref;
+		name = leftMostName ? leftMostName + "." + name : name;
+		originalName = leftMostName
+			? leftMostName + "." + originalName
+			: originalName;
+
+		if (potentialChunk && potentialChunk.renamed) {
+			originalName = originalName.replace(
+				"." + potentialChunk.renamed + ".",
+				"." + potentialChunk.original + "."
+			);
+		}
+	}
+
+	return {
+		renamedChain: name.split("."),
+		originalChain: originalName.split("."),
+	};
+}
+
+/**
+ * Gets enclosing scope
+ * 
+ * @param {Node} node 
+ * @returns {Scope}
+ */
+ function getEnclosingVariableScope (node) {
+	// Get to the nearest upper scope
+	let nearestScopeableNode = node;
+	while (
+		nearestScopeableNode && nearestScopeableNode.type &&
```

Comment:
Probably not the full list as for example `let` and `const` are block-scoped, but for some reason I'm not able to derive the scope in such cases. Probably would need to check if this is an issue or missing functionality in `escope`

Reply:
Now should be better : )
Thanks a lot!